src/membership/members.rs

```diff
@@ -485,4 +557,145 @@ impl<T: Trait> Module<T> {
         <MemberProfile<T>>::insert(id, profile);
         Ok(())
     }
+
+    // Checks if a member identified by their controller account occupies a role at least once
+    pub fn member_is_in_role(controller_account: &T::AccountId, role: Role) -> bool {
+        if let Ok(member_id) = Self::ensure_is_member_controller_account(controller_account) {
+            Self::ensure_profile(member_id).ok().map_or_else(
```

Comment:
This profile lookup cannot fail, but this code reads as it can. Write with an assertion that it's not possible.